src/main/java/io/pravega/perf/WriterWorker.java

```diff
@@ -57,67 +74,120 @@ public Void call() throws InterruptedException, ExecutionException, IOException
+            final EventsController eCnt = new EventsController(System.currentTimeMillis(), eventsPerSec);
+            for (int i = 0; i < events; i++) {
+                recordWrite(payload, stats::recordTime);
+                eCnt.control(i);
+            }
+            flush();
+        }
+    }
 
+    private class EventsWriterRW implements Performance {
+
+        public void benchmark() throws InterruptedException, IOException {
+            final EventsController eCnt = new EventsController(System.currentTimeMillis(), eventsPerSec);
             for (int i = 0; i < events; i++) {
+                final String val = System.currentTimeMillis() + ", " + workerID + ", ";
+                final String data = (val + payload).substring(0, messageSize);
+                writeData(data);
+                eCnt.control(i);
+            }
+            flush();
+        }
+    }
 
-                // Construct event payload
-                String val = System.currentTimeMillis() + ", " + workerID + ", " + (int) (Math.random() * 200);
-                String payload = String.format("%-" + messageSize + "s", val);
-                String key;
-                if (isRandomKey) {
-                    key = Integer.toString(workerID + rand.nextInt());
-                } else {
-                    key = Integer.toString(workerID);
-                }
+    private class EventsWriterTime implements Performance {
 
-                final Instant startTime = Instant.now();
-                retFuture = writeData(key, payload);
-                // event ingestion
-                retFuture = stats.recordTime(retFuture, startTime, payload.length());
-                stats.print();
-                tput.control(stats.eventsRate());
+        public void benchmark() throws InterruptedException, IOException {
+            final long msToRun = secondsToRun * MS_PER_SEC;
+            long time = System.currentTimeMillis();
+            final EventsController eCnt = new EventsController(time, eventsPerSec);
 
+            for (int i = 0; (time - startTime) < msToRun; i++) {
+                time = recordWrite(payload, stats::recordTime);
+                eCnt.control(i);
             }
-
             flush();
+        }
+    }
 
-            //Wait for the last packet to get acked
-            retFuture.get();
+    private class EventsWriterTimeRW implements Performance {
+
+        public void benchmark() throws InterruptedException, IOException {
+            final long msToRun = secondsToRun * MS_PER_SEC;
+            long time = System.currentTimeMillis();
+            final EventsController eCnt = new EventsController(time, eventsPerSec);
+
+            for (int i = 0; (time - startTime) < msToRun; i++) {
+                time = System.currentTimeMillis();
+                final String val = time + ", " + workerID + ", ";
+                final String data = (val + payload).substring(0, messageSize);
```

Comment:
Similar comment to the previous one regarding the payload.

Reply:
yes, its same as above.